File: backend/src/main/java/entities/Transacao.java
package entities;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@Entity
public class Transacao {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String tipo; // "DEBITO" ou "CREDITO"
    private Double valor;

    @ManyToOne
    @JoinColumn(name = "cliente_id")
    private Cliente cliente;
}

File: frontend/src/app/models/transacao.ts
import { Cliente } from './cliente';

export interface Transacao {
  id?: number;
  tipo: TipoTransacaoEnum;
  valor: number;
  cliente: Cliente;
}

export interface TransacaoDTO {
  tipo: string;
  valor: number;
  cliente: Partial<Cliente>;
}
export enum TipoTransacaoEnum {
  DEBITO = 'DEBITO',
  CREDITO = 'CREDITO',
}
